### PR TITLE
HTML: BroadcastChannel and SharedArrayBuffer success case revamp

### PR DESCRIPTION
**Pull request by annevk**
The original test could not work as far as I could tell. This should be able to work and also adds COOP/COEP headers for future proofing. No browser passes still.

### CODE REVIEW

**Comment by annevk** [comment]
Hmm, these also have to be HTTPS.

File: html/infrastructure/safe-passing-of-structured-data/shared-array-buffers/broadcastchannel-success.https.html [hunk 1]
```diff
@@ -6,30 +6,27 @@
...
   ]).then(() => {
     const sab = new SharedArrayBuffer(3);
     const view = new Uint8Array(sab);
-    const channel = new BroadcastChannel("channel name");
+    const channel = new BroadcastChannel(channelName);
 
     return new Promise(resolve => {
       let soFar = 0;
-      channel.onmessage = t.step_func(({ data: { sab: broadcastSAB, i } }) => {
-        if (broadcastSAB) {
```
Comment (domenic) [inline html/infrastructure/safe-passing-of-structured-data/shared-array-buffers/broadcastchannel-success.https.html]: Why is removing this OK?

Comment (annevk) [reply html/infrastructure/safe-passing-of-structured-data/shared-array-buffers/broadcastchannel-success.https.html]: Because the current channel is not a destination and none of the destinations transmit a sab.

File: html/infrastructure/safe-passing-of-structured-data/shared-array-buffers/broadcastchannel-success.https.html [hunk 2]
```diff
@@ -6,30 +6,27 @@
...
 <script src="/resources/testharnessreport.js"></script>
+<script src="/common/utils.js"></script>
 
 <div id="log"></div>
 
 <script>
 "use strict";
 
 promise_test(t => {
+  const channelName = token();
```
Comment (domenic) [inline html/infrastructure/safe-passing-of-structured-data/shared-array-buffers/broadcastchannel-success.https.html]: Why is this necessary? Is it guarding against running tests in multiple tabs at once? If so that'd be good to comment on, because for the purpose of our usual style of running a single test it doesn't seem necessary...

**Review by domenic — APPROVED** [review]
LGTM with clarity suggestion. The previous version was pretty broken.

I haven't been keeping up with the header discussions so I'm assuming you got those right.

File: html/infrastructure/safe-passing-of-structured-data/shared-array-buffers/broadcastchannel-success.https.html (hunk 2 above)
Comment (annevk) [reply html/infrastructure/safe-passing-of-structured-data/shared-array-buffers/broadcastchannel-success.https.html]: One of your colleagues gave the feedback of preferring this over trying to avoid collisions manually, e.g., when adding more tests. Seems reasonable to me and also helps with running tests in parallel.

Comment (domenic) [reply html/infrastructure/safe-passing-of-structured-data/shared-array-buffers/broadcastchannel-success.https.html]: I just don't understand why it's necessary at all in this case. There are not multiple tests here. If we are trying to support running multiple tests in parallel then we should note that explicitly since it's a rather unusual requirement.

Comment (annevk) [reply html/infrastructure/safe-passing-of-structured-data/shared-array-buffers/broadcastchannel-success.https.html]: I was mostly just trying to make it easier for someone adding a new test. And reportedly this helps Safari and Servo.

Comment (domenic) [reply html/infrastructure/safe-passing-of-structured-data/shared-array-buffers/broadcastchannel-success.https.html]: Seems reasonable, but not obvious from looking at the code, thus my suggestion to add a comment.

Comment (jgraham) [reply html/infrastructure/safe-passing-of-structured-data/shared-array-buffers/broadcastchannel-success.https.html]: In general the runner does support running multiple tests in parallel. Servo certainly uses this configuration in CI. Browsers with the concept of a profile typically use one profile per parallel instance, but e.g. Safari doesn't have the concept of profiles. By default we actually run all the tests in the same directory in the same parallel instance, but that's a customisable feature.
